### ISSUE TITLE
arrow functions prevent consideration of function objects

### ISSUE THREAD

**Issue by jeffreysbrother** [t9md/atom-vim-mode-plus#986]
The command `cif` (as well as other related actions such as `vaf` and `daf`) does not recognize the following function (presumably because it utilizes an ES6 arrow function):

```js
$(window).on('focus', () => {
	document.title = originialTitle;
	clearTimeout(tabTimeout);
	switched = false;
});

```

If the arrow function is replaced with a standard (pre-ES6) anonymous function, the issue does not arise. *However*, even if no anonymous function is used, it will still fail if the body of the function is not indented.

```js
$(window).on('focus', function () {
document.title = originialTitle;
clearTimeout(tabTimeout);
switched = false;
});

```

vim-mode-plus: 1.22.0
Atom: 1.22.1
Electron: 1.6.15
Chrome: 56.0.2924.87
Node: 7.4.0
MacOS: 10.12.6

**Comment by t9md** [comment]
@jeffreysbrother

You are right, and what you pointed out behaviors are both I already recognized.

### For `a f`, `i f` cannot select arrow function

Once it's possible, but I excluded it, because

- I want to select enclosing outer function rather than arrow function.
- Why I didn't excluded unnamed function is just because it's difficult to distinguish from named function.

I understand that it's inconsistent.
But to select arrow function, you can use fold text object as long as it's foldable arrow function.
By such as `y z`(short hand of `y a z`), `y a z`, `y i z` etc..

### For `a f`, `i f` should be able to select function regardless of how it's indented.

I agree, but currently vmp find function range from fold, fold is created based on indent.
I have no plan to change this for now.


### vmp provide very basic function text-object, and want to avoid language specific handling

Following are what I wrote in recent TextObject Function improvement PR. #984

> I won't try to be perfectly detect function body by parsing.
I will try to improve coverage of detection by supporting common multi-func-patterns used in different function.
I want keep language specific handling as minimum as possible.
So I will still get hint from indentation.
